### DISCUSSION TITLE
How to call a method in a button from an screen in another layout or screen?

### DISCUSSION THREAD

**Discussion by AkmalArzhang** [orchidsoftware/platform#1939] [Q&A]
Hi,

I would like to call a method/function in an screen in another screen on button method. For instance, in EditScreen I have a remove function, while in ListScreen I have a delete button that should call the remove function from EditScreen. How do I do that?

**Comment by gruz** [comment]
As far as I can understand it's not possible out of the box.
I'd use a trait with the method in several screens.

**Comment by lordjack** [comment]
I do not know if I could understand what you want @AkmalArzhang, but to call a button on the form or the listing you can do so.

For the form, you can do this: 
```php
class UserListScreen extends Screen
{
public function commandBar(): array
    {
        return [
            Link::make('Novo')
                ->icon('pencil')
                ->route('platform.user.edit')
        ];
    }
}
```

For the DataGrid you can do this: 
```php
class UserListLayout extends Table
{
protected function columns(): array
    {
        return [TD::make(__(''))
                ->align(TD::ALIGN_CENTER)
                ->width('100px')
                ->render(function (User $user) {
                    //   dd($user->id);
                    return
                        Link::make(__('Editar'))
                            ->route('platform.user.edit', $user->id)
                            ->icon('pencil');
                }),
                    ];
    }
 }
```
